cloud_Azure/terraform/module/examples/multiple_accounts_multiple_resource_group/azure_onboarder.py

```diff
@@ -0,0 +1,220 @@
+import argparse
+import configparser
+import logging
+import os
+import sys
+from dataclasses import dataclass
+from typing import Any, Callable, Dict, List, Tuple
+
+from az.cli import az
+from python_terraform import IsFlagged, IsNotFlagged, Terraform
+
+log = logging.getLogger(__name__)
+logging.basicConfig(format="%(levelname)s: %(message)s", level=logging.WARNING)
+EX_OK: int = 0  # exit code for successful command
+EX_FAILED: int = 1  # exit  code for failed command
+PROFILES_FILE: str = "profiles.ini"
+
+
+# pylint: disable=too-many-instance-attributes
+@dataclass
+class AzureProfile:
+    name: str
+    subscription_id: str
+    tenant_id: str
+    principal_id: str
+    principal_secret: str
+    location: str
+    resource_group_names: List[str]
+    storage_account_names: List[str]
+
+
+TerraformVars = Dict[str, Any]  # variables passed in terraform plan/apply/destroy call
+TerraformAction = Callable[[Terraform, TerraformVars], bool]  # terraform plan/apply/destroy
+
+# execute an action for every profile
```

Comment:
Please put it into function docstring. In other places too.

Reply:
updated

Reply:
Done